### PR TITLE
Define automatic labels

### PR DESCRIPTION
**Pull request by Skeptor**
This PR is the development for this issue: #1004
Now you can write a defined string for some labels and the agent will replace the information with your current environment configuration.
The labels start with "$(" and end with ")"
Current labels available are:

- OS Name: os.name
- OS Version: os.version
- IPv4 address from the default gateway: ipv4.primary
- IPv6 address from the default gateway: ipv6.primary
- The others IPv4 addresses: ipv4.others
- The others IPv6 addresses: ipv6.others
- The timezone of the environment: timezone
- Hostname of the environment: hostname

### CODE REVIEW

**Comment by Skeptor** [comment]
We're going to parse the labels in JSON format to get JSON objects.

**Comment by Skeptor** [comment]
Issue:
Some alerts in the Manager doesn't have the string replaced with the label information.

File: src/wazuh_modules/syscollector/syscollector_linux.c [hunk 1]
```diff
@@ -1239,33 +1212,70 @@ void sys_network_linux(int queue_fd, const char* LOCATION){
         cJSON_AddStringToObject(ipv6, "DHCP", dhcp_status);
         free(dhcp_status);
 
-        /* Send interface data in JSON format */
-        string = cJSON_PrintUnformatted(object);
-        mtdebug2(WM_SYS_LOGTAG, "sys_network_linux() sending '%s'", string);
-        wm_sendmsg(usec, queue_fd, string, LOCATION, SYSCOLLECTOR_MQ);
-        cJSON_Delete(object);
+        cJSON_AddItemToArray(ifaces_list_json,interface);
```
Comment (vikman90) [inline src/wazuh_modules/syscollector/syscollector_linux.c]: We are adding `interface` instead of `object`, so we are leaking `object`.

**Comment by Skeptor** [comment]
After executing analysisd with Valgrind there is still one memory leak. This is the valgrind trace:
```
150 bytes in 10 blocks are definitely lost in loss record 157 of 376
==13933==    at 0x4C2FB0F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==13933==    by 0x5B109B9: strdup (strdup.c:42)
==13933==    by 0x1AB40A: labels_add (labels_op.c:43)
==13933==    by 0x1587A3: Read_Labels (labels-config.c:77)
==13933==    by 0x14C978: read_main_elements (config.c:141)
==13933==    by 0x14CEB1: ReadConfig (config.c:228)
==13933==    by 0x11219C: GlobalConf (config.c:97)
==13933==    by 0x12BAE1: main (analysisd.c:368)
```
I still don't know surely what is the problem, but I think that it is in a weird behaviour in labels_add function (labels_op.c:22)
```
 size_t i;

    if (overwrite) {
        for (i = 0; labels && labels[i].key; i++) {
            if (!strcmp(labels[i].key, key)) {
                break;
            }
        }
    } else {
        i = *size;
    }

    if (!labels || i == *size) {
        os_realloc(labels, (*size + 2) * sizeof(wlabel_t), labels);
        labels[(*size)++].key = strdup(key);
        memset(labels + *size, 0, sizeof(wlabel_t));
    } else if (labels) {
        free(labels[i].value);
    }

    labels[i].value = strdup(value);
    labels[i].flags.hidden = hidden;
    return labels;
```
I think the problem is that line 43:
`    labels[i].value = strdup(value);`
Is being executed with labels[i].value not empty cause the flow is skipping the free in the conditional.

**Comment by Skeptor** [comment]
This feature is going to have a redesign:
The information is going to be extracted in the manager instead of being send by the agent. When the manager receive an event that trigger a rule it will use the information stored in the agent's DB.
The information required will be stored in memory and updated every hour.
This way the configuration is reserved to the manager and the process is more efficient.

**Pull request closed without merging**